enable text-overflow property for Servo

RichardTjokroutomo opened the pull request:
Enable `text-overflow` for Servo. Companion PR: [#40526](https://github.com/servo/servo/pull/40526)

delan committed:
Servo initial downstream commit

Any ancestors of this commit are from upstream mozilla-central, with
some filtering and renaming. Our patches and sync tooling start here.

The sync tooling has all been squashed into this commit, based on:
https://github.com/servo/stylo/commits/64731e10dc8ef87ef52aa2fb9f988c3b2530f3a7

delan committed:
Commit our changes on top of upstream Stylo

This is a rebase of d99b4a52f71b423363626d58506a731f3d06ccf6

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Sort static prefs by name (#303)

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

delan committed:
Commit our changes on top of upstream Stylo

This is a rebase of ab75cec2f1fdb1c6d5731b4c685a3bfc798f9619

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Loirooriol reviewed (changes requested):
As said in the Servo review, since it's not handling strings or 2 values, this should alter the parser to reject these cases for Servo.

Alternatively, you can put this property behind a flag, and then in Servo make it an experimental web platform feature flag. Then it won't be enabled by default, but it will be used in tests.

Loirooriol reviewed (commented):
Since changing the struct affects things more than what I expected, maybe just do something like

```rs
        #[cfg(feature = "servo")]
        if matches!(first, TextOverflowSide::String(_)) {
            return Err(input.new_custom_error(StyleParseErrorKind::UnspecifiedError));
        }
        #[cfg(feature = "gecko")]
        if let Ok(second) = input.try_parse(|input| TextOverflowSide::parse(context, input)) {
            return Ok(Self {
                first,
                second,
                sides_are_logical: false,
            });
        }
        Ok(Self {
            first: TextOverflowSide::Clip,
            second: first,
            sides_are_logical: true,
        })
```

Or add support for strings as Nico says in https://github.com/servo/servo/pull/40526#issuecomment-3581285687

RichardTjokroutomo committed:
enable text overflow

Signed-off-by: Richard Tjokroutomo <richard.tjokro2@gmail.com>

Loirooriol committed:
Fixup for https://phabricator.services.mozilla.com/D280938

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Loirooriol committed:
Fixup for https://phabricator.services.mozilla.com/D281332

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Loirooriol committed:
Fixup for https://phabricator.services.mozilla.com/D280243

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Loirooriol committed:
Fixup for https://phabricator.services.mozilla.com/D282398

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Loirooriol committed:
Fixup for https://phabricator.services.mozilla.com/D284054

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Loirooriol committed:
Fixup for https://phabricator.services.mozilla.com/D284378

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Loirooriol committed:
Fixup for https://phabricator.services.mozilla.com/D284172

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Loirooriol committed:
Fixup for https://phabricator.services.mozilla.com/D284297

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

Loirooriol committed:
Bump versions
- Bump stylo_* to 0.13.0
- Bump selectors to 0.36.0

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

TimvdLippe committed:
Add beforeinput as atom (#326)

Required for firing events for contenteditable

Signed-off-by: Tim van der Lippe <tvanderlippe@gmail.com>

Loirooriol committed:
Add logic for registering custom properties from script (#325)

This way it can be shared among Gecko and Servo.

Stylo PR: https://github.com/servo/servo/pull/43085
Upstream: https://bugzilla.mozilla.org/show_bug.cgi?id=2021743

Signed-off-by: Oriol Brufau <obrufau@igalia.com>

stevennovaryo committed:
Make `::placeholder` public and add property restriction for `::placeholder` and `::marker` (#322)

The `::placeholder` pseudo element should be a public pseudo element.
And, both `::placeholder` and `::marker` should have a property
restriction as defined in the spec. In stylo, the property restriction
has been computed in `PropertyFlags`.

Servo PR: https://github.com/servo/servo/pull/43053

Signed-off-by: stevennovaryo <steven.novaryo@gmail.com>

yezhizhen committed:
Remove parse condition `step_position_jump_enabled` (#323)

This enables `jump-start`, `jump-end`, `jump-none`, `jump-both`.

The feature is left here 7 years ago. As tested with examples, all
`step-easing-function`s work as expected.

Servo PR: https://github.com/servo/servo/pull/43061

---------

Signed-off-by: Euclid Ye <yezhizhenjiakang@gmail.com>
Signed-off-by: Euclid Ye <yezhizhenjiakang@gmail.com>>

mrobinson committed:
Restore Servo's original form control theme colors (#327)

This change does two things:
1. Restores Servo's original theme colors from before support for system
color was added.
2. Eliminates the selection of default system color based on the color
scheme of the system. This code was dead code since
`is_dark_color_scheme` always returns `false`.

Servo PR: servo/servo#43107

Signed-off-by: Martin Robinson <mrobinson@igalia.com>

Loirooriol committed:
Share `Device` logic among Gecko and Servo (#330)

Adds a new file `style/device/mod.rs` with the `Device` logic that is
common for Gecko and Servo.

The Gecko-specific logic for `Device` is moved into
`style/device/gecko.rs`, and the previous `style/gecko/media_queries.rs`
is removed.

The Servo-specific logic for `Device` is moved into
`style/device/servo.rs`. The previous `servo/media_queries.rs` also had
logic for media features, which is now moved into
`style/servo/media_features.rs` for consistency with Gecko.

Servo PR: https://github.com/servo/servo/pull/43146

Signed-off-by: Oriol Brufau <obrufau@igalia.com>
